jerry-ext/arg/arg-transform-functions.c

```diff
@@ -84,14 +85,91 @@ jerryx_arg_transform_number (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
                                (jerry_char_t *) "It can not be converted to a number.");
   }
 
-  double *dest = c_arg_p->dest;
-  *dest = jerry_get_number_value (to_number);
+  *number_p = jerry_get_number_value (to_number);
   jerry_release_value (to_number);
 
   return jerry_create_undefined ();
+} /* jerryx_arg_transform_number_common */
+
+/**
+ * clamp a double by given min and max value
+ * @return clamped double value
+ */
+static double
+jerryx_arg_clamp_double (double d, /**< the double value */
+                         double min, /**< min value */
+                         double max) /**< max value */
+{
+  const double t = d < min ? min : d;
+  return t > max ? max : t;
+} /* jerryx_arg_clamp_double */
+
+/**
+ * Tranform a JS argument to a double. Type coercion is not allowed.
+ *
+ * @return jerry undefined: the transformer passes,
+ *         jerry error: the transformer fails.
+ */
+jerry_value_t
+jerryx_arg_transform_number_strict (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< available JS args */
+                                    const jerryx_arg_t *c_arg_p) /**< the native arg */
+{
+  return jerryx_arg_transform_number_strict_common (js_arg_iter_p,
+                                                    c_arg_p->dest);
+} /* jerryx_arg_transform_number_strict */
+
+/**
+ * Tranform a JS argument to a double. Type coercion is allowed.
+ *
+ * @return jerry undefined: the transformer passes,
+ *         jerry error: the transformer fails.
+ */
+jerry_value_t
+jerryx_arg_transform_number (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< available JS args */
+                             const jerryx_arg_t *c_arg_p) /**< the native arg */
+{
+  return jerryx_arg_transform_number_common (js_arg_iter_p,
+                                             c_arg_p->dest);
 } /* jerryx_arg_transform_number */
 
 /**
+ * Use the macro to define two transform functions for int type.
+ * One is non-strict version and the other is the strict version.
+ */
+#define GENERATE_JERRYX_ARG_TRANSFORM_FUNC_FOR_INT(type, min, max) \
+jerry_value_t jerryx_arg_transform_ ## type ## _strict (jerryx_arg_js_iterator_t *js_arg_iter_p, \
+                                                        const jerryx_arg_t *c_arg_p) \
+{ \
+  double tmp; \
+  jerry_value_t rv = jerryx_arg_transform_number_strict_common (js_arg_iter_p, &tmp);\
+  if (!jerry_value_has_error_flag (rv)) \
+  { \
+    tmp = jerryx_arg_clamp_double (tmp, min, max); \
+    *(type ## _t *) c_arg_p->dest = (type ## _t) tmp; \
```

Comment:
I don't like the idea of "flattening" these 2 independent dimensions (rounding and out-of-range policies) onto 1 enum type.
I like your idea of using `extra_info` though!

What if "compress" the struct so that it fits into `extra_info`, by using bitfields?
```
typedef struct {
  jerryx_arg_coerce_t coercion:1;
  jerryx_arg_optional_t optionality:1;
  jerryx_arg_rounding_t rounding:2;
  jerryx_arg_clipping_t clipping:1;
} jerryx_arg_number_options_t;

__Static_assert(sizeof(jerryx_arg_number_options_t) <= sizeof(((jerryx_arg_t *)0)->extra_info), "jerryx_arg_number_options_t must fit into extra_info");
```

Additional benefit: I think this way we can have just *one* internal transform function for all the int types.